lib/charms/grafana_agent/v0/cos_agent.py

```diff
@@ -91,9 +91,25 @@ def __init__(self, *args):
             self,
             relation_name="custom-cos-agent",
             metrics_endpoints=[
+                # specify "path" and "port" to scrape from localhost
                 {"path": "/metrics", "port": 9000},
                 {"path": "/metrics", "port": 9001},
                 {"path": "/metrics", "port": 9002},
+
+                # OR, specify a full "scrape_job" definition, e.g.
+                {
+                    "scrape_job": {
+                        "job_name": "custom_job",
+                        "metrics_path": "/metrics",
+                        "authorization": {"credentials": "bearer-token"},
+                        "static_configs": [
+                            {
+                                "targets": ["localhost:9000"]},
+                                "labels": {"key": "value"},
+                            },
+                        ],
+                    },
+                }
             ],
```

Comment:
I like this.
1. I'm not a fan of "implicit dual schema". I'd rather bump LIBAPI and only have the "scrape_job" option.
2. Iiuc, we could probably omit the "scrape_job" key (it's actually "[scrape_config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config)"). 

```suggestion
            scrape_configs=[
                {
                        "job_name": "custom_job",
                        "metrics_path": "/metrics",
                        "authorization": {"credentials": "bearer-token"},
                        "static_configs": [
                            {
                                "targets": ["localhost:9000"]},
                                "labels": {"key": "value"},
                            },
                        ],
                },
            ],
```
Wdyt?
cc: @simskij @Abuelodelanada @dstathis @lucabello @PietroPasotti

Reply:
I'm 100% in on this, I understand the `metrics_endpoints` was only added for simplicity. Let me know how to proceed

Reply:
Wdyt about this [commit](https://github.com/canonical/grafana-agent-k8s-operator/pull/220/commits/da96b8f005cbb08e73234387fd7c0fb15e4a7230)?

Reply:
After merging one of your other PRs we now have a conflict with the Callable.
I'm leaving this for you to resolve as I'm not sure which solution best fits your use case.

Reply:
> Wdyt about this [commit](https://github.com/canonical/grafana-agent-k8s-operator/pull/220/commits/da96b8f005cbb08e73234387fd7c0fb15e4a7230)?

Looks great, let me get back to this to fix the conflicts and I'll ping you. Thanks a lot!

Reply:
Updated, but left a comment elsewhere. I like being explicit with a `scrape_configs` for this :+1: